builder/frameworks/ulp_lp_core.py

```
@@ -0,0 +1,445 @@
+#
+# sdkconfig resolution
+#
+# Priority: user-provided ulp/sdkconfig.h > recompiled from libs (validated above)
+#
+
+USER_SDKCONFIG_H = ULP_DIR / "sdkconfig.h"
+
+
+def get_sdkconfig_value(key, default):
+    try:
+        custom = env.GetProjectOption("custom_sdkconfig", "")
```

Ruff (0.15.5)

[error] Undefined name `env`

(F821)